dll/win32/shell32/droptargets/CFSDropTarget.cpp

```diff
@@ -588,12 +588,19 @@ HRESULT CFSDropTarget::_DoDrop(IDataObject *pDataObject,
             /* We need to create a link for each pidl in the copied items, so step through the pidls from the clipboard */
             for (UINT i = 0; i < lpcida->cidl; i++)
             {
+                SFGAOF att = SHGetAttributes(psfFrom, apidl[i], SFGAO_FOLDER | SFGAO_STREAM | SFGAO_FILESYSTEM);
                 CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidlFull;
                 hr = SHILCombine(pidl, apidl[i], &pidlFull);
 
                 WCHAR targetName[MAX_PATH];
                 if (SUCCEEDED(hr))
-                    hr = Shell_DisplayNameOf(psfFrom, apidl[i], SHGDN_FOREDITING | SHGDN_INFOLDER, targetName, _countof(targetName));
+                {
+                    // If the target is a file, we use SHGDN_FORPARSING because "NeverShowExt" will hide the ".lnk" extension.
```

Comment:
not possible to calculate the targetname first (see below) and just do a `!wcsicmp(PathFindExtensionW(targetName), L".lnk")` check to see whether we're dealing with a link file?

Reply:
I'm not sure I follow. We could just ask for SHGDN_FORPARSING and check for ".lnk" yes but we then have to call it again asking for SHGDN_FOREDITING in other cases, otherwise we end up with names like "{GUID}.lnk" for virtual items.

We are already asking for the attributes anyway so the UseParsing test is basically free.